OpenSim/Actuators/MuscleFirstOrderActivationDynamicModel.cpp

```diff
@@ -160,14 +79,38 @@ calcDerivative(double activation, double excitation) const
 {
     // This model respects a lower bound on activation while preserving the
     // expected steady-state value.
-    double clampedExcitation = clamp(getMinimumActivation(), excitation, 1.0);
-    double clampedActivation = clamp(getMinimumActivation(), activation, 1.0);
+    double clampedExcitation = clamp(get_minimum_activation(), excitation, 1.0);
+    double clampedActivation = clamp(get_minimum_activation(), activation, 1.0);
     double tau = SimTK::NaN;
 
     if(clampedExcitation > clampedActivation) {
-        tau = getActivationTimeConstant() * (0.5 + 1.5*clampedActivation);
+        tau = get_activation_time_constant() * (0.5 + 1.5*clampedActivation);
     } else {
-        tau = getDeactivationTimeConstant() / (0.5 + 1.5*clampedActivation);
+        tau = get_deactivation_time_constant() / (0.5 + 1.5*clampedActivation);
     }
     return (clampedExcitation - clampedActivation) / tau;
 }
+
+//==============================================================================
+// COMPONENT INTERFACE
+//==============================================================================
+void MuscleFirstOrderActivationDynamicModel::extendFinalizeFromProperties()
+{
+    Super::extendFinalizeFromProperties();
+
+    // Ensure property values are within appropriate ranges.
+    SimTK_ERRCHK1_ALWAYS(get_activation_time_constant() >= SimTK::SignificantReal,
+        "MuscleFirstOrderActivationDynamicModel::extendFinalizeFromProperties",
+        "%s: Activation time constant can be no smaller than SimTK::SignificantReal",
+        getName().c_str());
+    SimTK_ERRCHK1_ALWAYS(get_deactivation_time_constant() >= SimTK::SignificantReal,
+        "MuscleFirstOrderActivationDynamicModel::extendFinalizeFromProperties",
+        "%s: Deactivation time constant can be no smaller than SimTK::SignificantReal",
+        getName().c_str());
+    SimTK_ERRCHK1_ALWAYS(get_minimum_activation() >= 0.0
+        && get_minimum_activation() <= 1.0-SimTK::SignificantReal,
+        "MuscleFirstOrderActivationDynamicModel::extendFinalizeFromProperties",
+        "%s: Minimum activation must be in the range [0,1)", getName().c_str());
+
+    setObjectIsUpToDateWithProperties();
```

Comment:
@chrisdembia file an issue. More likely than not, SO shouldn't need to make direct use.